src/ImGui.NET/ImGui.Manual.cs

```diff
@@ -355,6 +355,47 @@ public static unsafe partial class ImGui
             return result != 0;
         }
 
+        public static Vector2 CalcTextSize(
+            string text,
+            int? start = null,
+            int? length = null,
+            bool hideTextAfterDoubleHash = false,
+            float wrapWidth = -1.0f)
+        {
+            Vector2 ret;
+            byte* nativeText = null;
+            byte* nativeTextStart = null;
+            byte* nativeTextEnd = null;
+            int textByteCount = 0;
+            int textByteSize = 0;
+            if (text != null)
+            {
+                textByteCount = Encoding.UTF8.GetByteCount(text);
+                textByteSize = Encoding.UTF8.GetByteCount("X");
```

Comment:
The usage of this variable doesn't seem correct to me. "X" will always have a 1-byte UTF8 representation, so computing it here seems unnecessary. But not all C# chars will have a 1-byte representation, particularly in other languages. I'm not sure exactly how to address this, presumably you would need to scan through the string to find the offset of the particular element you need.

Reply:
Agree, this isn't perfect/currect.

->  you would need to scan through the string to find the offset of the particular element you need.

assuming `you` means caller over here, rather than accepting `start` and `length`, the function can accept `startInByteSize` and `lengthInByteSize` to put the burden on the caller to figure out the `startOffsetInByte` and `LengthOffsetInByte`.

Reply:
I think the current signature of this function is correct, with `start` and `length` referring to C# char indices. This is what 99.9% of users will expect, because all other string-related functions using "start" and "length" behave that way, so I don't think we should push anything onto the caller. However, when converting to UTF8 and UTF8-based offsets, we might need to do something different with the computed addresses, because they aren't necessarily 1-to-1 with char indices.

Reply:
Looks like it's not an easy problem to solve. I am open to suggestions/code-changes. Following is the best I can come up with.

https://gist.github.com/zaafar/ccf2c32a9fa4d85cbe0dcd7163beb81d#file-imgui-manual-cs-L46

NOTE: The code is in public gist because I don't want to pollute the PR with potential solution. Sorry about that.

Reply:
Drawback (that I can think of): ReadOnlySpan is not available in .net standard 2.0.

Reply:
other than using SubString(start, length) or AsSpan(start, length), I couldn't figure out any way to convert `C# char indices` to `UTF8 based offsets`.

Reply:
1 benefit is that, only the portion of the string that needs to be computed by `CalcTextSize` is allocated via `Util.Allocate`/`Util.Allocate` function.

Reply:
never mind, found a solution that is better and works. so done.